src/util/ThreadPoolDescriptor.cpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2023      Alex Biddulph <bidskii@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "ThreadPoolDescriptor.hpp"
+
+namespace NUClear {
+namespace util {
+
+    const uint64_t ThreadPoolIDSource::MAIN_THREAD_POOL_ID    = 0;
+    const uint64_t ThreadPoolIDSource::DEFAULT_THREAD_POOL_ID = 1;
```

Comment:
I think these can go in the header if they are constexpr?
Unless there is a linking problem somewhere where they've been taken by reference.
In the right context you can assign them to a variable and then pass them in to avoid that too I think.

Reply:
I moved them here because there are undefined references with some compilers

Reply:
even with constexpr?

Reply:
These used to be `static constexpr ....` with an inline initialiser in the header, certain compilers hated it and complained about undefined references